src/Agent.PluginCore/PluginUtil.cs

```diff
@@ -0,0 +1,1611 @@
+
+            // Remove Invariant from the list of accepted languages.
+            //
+            // The constructor of VssHttpRequestSettings (base class of VssClientHttpRequestSettings) adds the current
+            // UI culture to the list of accepted languages. The UI culture will be Invariant on OSX/Linux when the
+            // LANG environment variable is not set when the program starts. If Invariant is in the list of accepted
+            // languages, then "System.ArgumentException: The value cannot be null or empty." will be thrown when the
+            // settings are applied to an HttpRequestMessage.
+            settings.AcceptLanguages.Remove(CultureInfo.InvariantCulture);
+
+            VssConnection connection = new VssConnection(serverUri, credentials, settings);
+            return connection;
+        }
+
+        public static VssCredentials GetVssCredential(ServiceEndpoint serviceEndpoint)
+        {
+            NotNull(serviceEndpoint, nameof(serviceEndpoint));
+            NotNull(serviceEndpoint.Authorization, nameof(serviceEndpoint.Authorization));
+            NotNullOrEmpty(serviceEndpoint.Authorization.Scheme, nameof(serviceEndpoint.Authorization.Scheme));
+
+            if (serviceEndpoint.Authorization.Parameters.Count == 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(serviceEndpoint));
+            }
+
+            VssCredentials credentials = null;
+            string accessToken;
+            if (serviceEndpoint.Authorization.Scheme == EndpointAuthorizationSchemes.OAuth &&
+                serviceEndpoint.Authorization.Parameters.TryGetValue(EndpointAuthorizationParameters.AccessToken, out accessToken))
+            {
+                credentials = new VssCredentials(null, new VssOAuthAccessTokenCredential(accessToken), CredentialPromptType.DoNotPrompt);
+            }
+
+            return credentials;
+        }
+        #endregion
+
+        #region IOUtil
+        public static T LoadObject<T>(string path)
+        {
+            string json = File.ReadAllText(path, Encoding.UTF8);
+            return ConvertFromJson<T>(json);
+        }
+
+        public static string GetBinPath()
+        {
+            return Path.GetDirectoryName(Assembly.GetEntryAssembly().Location);
+        }
+
+        public static void Delete(string path, CancellationToken cancellationToken)
+        {
+            DeleteDirectory(path, cancellationToken);
+            DeleteFile(path);
+        }
+
+        public static void DeleteDirectory(string path, CancellationToken cancellationToken)
+        {
+            DeleteDirectory(path, contentsOnly: false, continueOnContentDeleteError: false, cancellationToken: cancellationToken);
+        }
+
+        public static void DeleteDirectory(string path, bool contentsOnly, bool continueOnContentDeleteError, CancellationToken cancellationToken)
```

Comment:
consider moving this to shared DLL so we don't have to maintain a fork of the code. perhaps something to clean up in a following pr

Reply:
I agree, we need to figure out a good name.

Reply:
agent.listener.exe
agent.worker.exe
microsoft.vs.service.agent.dll
agent.pluginhost.exe
agent,plugins.dll
agent.sdk.dll

Reply:
fixed.